credentials/apps/credentials/utils.py

```
@@ -87,7 +172,46 @@ def get_program_api_client():
     return _get_service_user_api_client(programs_api_url, service_username, jwt_audience, jwt_secret_key)
 
 
-def _get_service_user_api_client(api_url, service_username, jwt_audience, jwt_secret_key):
+def get_organizations_api_client():
+    """
+    Return api client to communicate with the organizations api by using the
+    credentials service user in the LMS.
+    """
+    try:
+        organizations_api_url = settings.ORGANIZATIONS_API_URL
+        service_username = settings.CREDENTIALS_SERVICE_USER
+        jwt_audience = settings.ORGANIZATIONS_AUDIENCE
+        jwt_secret_key = settings.ORGANIZATIONS_SECRET_KEY
+    except AttributeError:
```

These settings are crucial to your service operating correctly. Realistically, the credentials service should NOT start if these settings are not available. If you simply catch/log the exception, you need to have monitoring in place for this log message. This means a few dozen users will see an error message before an alert is fired, someone notices the alert, the issue is investigated, and a solution finally put in place. That's a long time.

Let's say you added a check at startup. When we deploy to stage/prod the service doesn't start. If the service doesn't start we don't even switch the ELB. No users are affected. No time is wasted debugging.

You want to fail early. In some cases, catching/logging the exception would be fine. In this case, when the error is clearly associated with a person having the invalid settings, an error from which the software cannot recover without external intervention, you should not even start the program.